plugins/scorecards/src/api/ScorecardsClient.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ScorecardsApi } from './ScorecardsApi';
+import {
+  CheckResult,
+  CheckResponse,
+} from '@backstage/plugin-tech-insights-common';
+import { DiscoveryApi } from '@backstage/core-plugin-api';
+
+export type Options = {
+  discoveryApi: DiscoveryApi;
+  proxyPath?: string;
+};
+
+export class ScorecardsClient implements ScorecardsApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private baseUrl: string = '';
+
+  constructor(options: Options) {
+    this.discoveryApi = options.discoveryApi;
+  }
+
+  getBaseUrl: () => Promise<string> = async () => {
+    if (!this.baseUrl) {
+      this.baseUrl = await this.discoveryApi.getBaseUrl('tech-insights');
+    }
+    return this.baseUrl;
+  };
+
+  async fetchAllChecks(): Promise<CheckResult[]> {
+    const url = await this.getBaseUrl();
+    const allChecks = await fetch(`${url}/checks`);
+    const payload = await allChecks.json();
+    if (!allChecks.ok) {
+      throw new Error(payload.errors[0]);
+    }
+    const checks = payload.map((check: CheckResponse) => check.id);
+    return checks;
+  }
+
+  async fetchChecks({
```

Comment:
This should be called `runChecks` or similar IMO

Reply:
We should either create a variation of this that accepts `checkIds` array or just reuse this with an optional param with some logic to use the passed in array or retrieved `allChecks`.

Reply:
Wouldn't we have a 422 error if no checkIds passed?

Reply:
I mean checks passed into this method. As it is now it enforces users to use all checks in the system and that will end up being impossible to create different scorecards with different checks that way (without running all checks every time).

We should remove the 422 from that endpoint since the logic in the backend doesn't necessarily need it anymore.

Reply:
@Xantier Should I do it with this PR? I mean removing 422

Reply:
Let's do it on this one yeah